WhitespaceAfter should support LITERAL_FINALLY token

https://checkstyle.org/config_whitespace.html#WhitespaceAfter

I have downloaded the latest cli from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

**How it works Now:**

```bash
/var/tmp $ javac FinallyStatement.java
/var/tmp $

/var/tmp $ cat config.xml
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="WhitespaceAfter">
            <property name="tokens" value="LITERAL_FINALLY"/>
        </module>
    </module>
</module>

/var/tmp $ cat FinallyStatement.java
public class FinallyStatement {
    public static void main(String[] args) {
        try {
            System.out.println(Integer.parseInt(args[0]));
        } finally{
            System.out.println("Finished");
        }
    }
}


/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-10.1-all.jar -c config.xml FinallyStatement.java
com.puppycrawl.tools.checkstyle.api.CheckstyleException: cannot initialize module TreeWalker - Token "LITERAL_FINALLY" was not found in Acceptable tokens list in check com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAfterCheck
        at com.puppycrawl.tools.checkstyle.Checker.setupChild(Checker.java:476)
        at com.puppycrawl.tools.checkstyle.api.AutomaticBean.configure(AutomaticBean.java:201)
        at com.puppycrawl.tools.checkstyle.Main.runCheckstyle(Main.java:403)
        at com.puppycrawl.tools.checkstyle.Main.runCli(Main.java:330)
        at com.puppycrawl.tools.checkstyle.Main.execute(Main.java:189)
        at com.puppycrawl.tools.checkstyle.Main.main(Main.java:126)
Caused by: com.puppycrawl.tools.checkstyle.api.CheckstyleException: Token "LITERAL_FINALLY" was not found in Acceptable tokens list in check com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAfterCheck
        at com.puppycrawl.tools.checkstyle.TreeWalker.registerCheck(TreeWalker.java:223)
        at com.puppycrawl.tools.checkstyle.TreeWalker.setupChild(TreeWalker.java:133)
        at com.puppycrawl.tools.checkstyle.api.AutomaticBean.configure(AutomaticBean.java:201)
        at com.puppycrawl.tools.checkstyle.Checker.setupChild(Checker.java:471)
        ... 5 more
Checkstyle ends with 1 errors.
```

**Is your feature request related to a problem? Please describe.**
It would be nice if checkstyle could detect missing space after `finally` keyword.

**Describe the solution you'd like**
WhitespaceAfter could support new literal type - `LITERAL_FINALLY`

**Additional context**
Google Java Style Guide requires whitespace before _almost_ any `{` usage.
See https://google.github.io/styleguide/javaguide.html#s4.6.2-horizontal-whitespace
>3. Before any open curly brace (`{`), with two exceptions:
> `@SomeAnnotation({a, b})` (no space is used)
> `String[][] x = {{"foo"}};` (no space is required between `{{`, by item 9 below)

#8122 seems related. This feature request is part of it.

#11489 is similar request for `try` keyword support

--- a/src/it/java/com/google/checkstyle/test/chapter4formatting/rule462horizontalwhitespace/WhitespaceAfterTest.java
+++ b/src/it/java/com/google/checkstyle/test/chapter4formatting/rule462horizontalwhitespace/WhitespaceAfterTest.java
@@ -57,6 +57,8 @@ public class WhitespaceAfterTest extends AbstractGoogleModuleTestSupport {
             "23:26: " + getCheckMessage(clazz, message, "->"),
             "24:9: " + getCheckMessage(clazz, message, "switch"),
             "31:9: " + getCheckMessage(clazz, message, "try"),
+            "35:16: " + getCheckMessage(clazz, message, "finally"),
+            "36:38: " + getCheckMessage(clazz, message, "finally"),
         };
         final Configuration checkConfig = getModuleConfig("WhitespaceAfter");
         final String filePath = getPath("InputWhitespaceAfterBad.java");

--- a/src/it/resources/com/google/checkstyle/test/chapter4formatting/rule462horizontalwhitespace/InputWhitespaceAfterBad.java
+++ b/src/it/resources/com/google/checkstyle/test/chapter4formatting/rule462horizontalwhitespace/InputWhitespaceAfterBad.java
@@ -30,4 +30,9 @@ public class InputWhitespaceAfterBad {
     public void check4() throws java.io.IOException {
         try(java.io.InputStream ignored = System.in;) { } // warn
     }
+
+    public void check5() {
+        try {} finally{} // warn
+        try {} catch (Exception e){} finally{} // warn
+    }
 }

--- a/src/it/resources/com/google/checkstyle/test/chapter4formatting/rule462horizontalwhitespace/InputWhitespaceAfterGood.java
+++ b/src/it/resources/com/google/checkstyle/test/chapter4formatting/rule462horizontalwhitespace/InputWhitespaceAfterGood.java
@@ -31,4 +31,9 @@ public class InputWhitespaceAfterGood {
         try (java.io.InputStream ignored = System.in) {}
         try {} catch(Exception e){}
     }
+
+    public void check5() {
+        try {} finally {}
+        try {} catch (Exception e){} finally {}
+    }
 }

--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java
@@ -54,6 +54,8 @@ import com.puppycrawl.tools.checkstyle.utils.CommonUtil;
  * LITERAL_DO</a>,
  * <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_FOR">
  * LITERAL_FOR</a>,
+ * <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_FINALLY">
+ * LITERAL_FINALLY</a>,
  * <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_DO">
  * DO_WHILE</a>,
  * <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#ELLIPSIS">
@@ -92,6 +94,12 @@ import com.puppycrawl.tools.checkstyle.utils.CommonUtil;
  *
  *      try {} catch (Exception e){} // OK
  *      try{} catch (Exception e){} // violation ''try' is not followed by whitespace'
+ *
+ *      try {} finally {} // OK
+ *      try {} finally{} // violation ''finally' is not followed by whitespace'
+ *
+ *      try {} catch (Error e){} finally {} // OK
+ *      try {} catch (Error e){} finally{} // violation ''finally' is not followed by whitespace'
  *  }
  * </pre>
  * <p>
@@ -163,6 +171,7 @@ public class WhitespaceAfterCheck
             TokenTypes.LITERAL_WHILE,
             TokenTypes.LITERAL_DO,
             TokenTypes.LITERAL_FOR,
+            TokenTypes.LITERAL_FINALLY,
             TokenTypes.DO_WHILE,
             TokenTypes.ELLIPSIS,
             TokenTypes.LITERAL_SWITCH,

--- a/src/main/resources/com/puppycrawl/tools/checkstyle/meta/checks/whitespace/WhitespaceAfterCheck.xml
+++ b/src/main/resources/com/puppycrawl/tools/checkstyle/meta/checks/whitespace/WhitespaceAfterCheck.xml
@@ -11,7 +11,7 @@
  EmptyForIteratorPad&lt;/a&gt; to validate empty for iterators.
  &lt;/p&gt;</description>
          <properties>
-            <property default-value="COMMA,SEMI,TYPECAST,LITERAL_IF,LITERAL_ELSE,LITERAL_WHILE,LITERAL_DO,LITERAL_FOR,DO_WHILE,ELLIPSIS,LITERAL_SWITCH,LITERAL_TRY,LAMBDA"
+            <property default-value="COMMA,SEMI,TYPECAST,LITERAL_IF,LITERAL_ELSE,LITERAL_WHILE,LITERAL_DO,LITERAL_FOR,LITERAL_FINALLY,DO_WHILE,ELLIPSIS,LITERAL_SWITCH,LITERAL_TRY,LAMBDA"
                       name="tokens"
                       type="java.lang.String[]"
                       validation-type="tokenSet">

--- a/src/main/resources/google_checks.xml
+++ b/src/main/resources/google_checks.xml
@@ -105,7 +105,7 @@
     <module name="WhitespaceAfter">
       <property name="tokens"
                value="COMMA, SEMI, TYPECAST, LITERAL_IF, LITERAL_ELSE,
-                    LITERAL_WHILE, LITERAL_DO, LITERAL_FOR, DO_WHILE, ELLIPSIS,
+                    LITERAL_WHILE, LITERAL_DO, LITERAL_FOR, LITERAL_FINALLY, DO_WHILE, ELLIPSIS,
                     LITERAL_SWITCH, LITERAL_TRY, LAMBDA"/>
     </module>
     <module name="WhitespaceAround">

--- a/src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheckTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheckTest.java
@@ -129,6 +129,17 @@ public class WhitespaceAfterCheckTest
     }
 
     @Test
+    public void testLiteralFinally() throws Exception {
+        final String[] expected = {
+            "14:13: " + getCheckMessage(MSG_WS_NOT_FOLLOWED, "finally"),
+            "17:31: " + getCheckMessage(MSG_WS_NOT_FOLLOWED, "finally"),
+        };
+        verifyWithInlineConfigParser(
+            getPath("InputWhitespaceAfterLiteralFinally.java"),
+            expected);
+    }
+
+    @Test
     public void testLiteralDo() throws Exception {
         final String[] expected = {
             "70:9: " + getCheckMessage(MSG_WS_NOT_FOLLOWED, "do"),

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/whitespaceafter/InputWhitespaceAfterLiteralFinally.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/whitespaceafter/InputWhitespaceAfterLiteralFinally.java
@@ -1,0 +1,19 @@
+/*
+WhitespaceAfter
+tokens = LITERAL_FINALLY
+
+
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.whitespace.whitespaceafter;
+
+public class InputWhitespaceAfterLiteralFinally {
+    public static void main(String[] args) {
+
+     try {} finally {} // OK
+     try {} finally{} // violation ''finally' is not followed by whitespace'
+
+     try {} catch (Error e){} finally {} // OK
+     try {} catch (Error e){} finally{} // violation''finally' is not followed by whitespace'
+    }
+}

--- a/src/xdocs/config_whitespace.xml
+++ b/src/xdocs/config_whitespace.xml
@@ -2900,6 +2900,8 @@ class Bar {
                   LITERAL_DO</a>
                 , <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_FOR">
                   LITERAL_FOR</a>
+                , <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_FINALLY">
+                  LITERAL_FINALLY</a>
                 , <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_DO">
                   DO_WHILE</a>
                 , <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#ELLIPSIS">
@@ -2930,6 +2932,8 @@ class Bar {
                   LITERAL_DO</a>
                 , <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_FOR">
                   LITERAL_FOR</a>
+                , <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_FINALLY">
+                  LITERAL_FINALLY</a>
                 , <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_DO">
                   DO_WHILE</a>
                 , <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#ELLIPSIS">
@@ -2976,6 +2980,12 @@ public void myTest() {
 
     try {} catch (Exception e){} // OK
     try{} catch (Exception e){} // violation ''try' is not followed by whitespace'
+
+    try {} finally {} // OK
+    try {} finally{} // violation ''finally' is not followed by whitespace'
+
+    try {} catch (Error e){} finally {} // OK
+    try {} catch (Error e){} finally{} // violation ''finally' is not followed by whitespace'
 }
           </pre>
         </div>

